### ISSUE TITLE
Throw overflow exception or upgrade LList API to enforce 255 limit

### ISSUE THREAD

**Issue by Noggog** [Mutagen-Modding/Mutagen#77]
Right now the error happens in the mod write phase with an overflow exception when trying to fit into a byte.  Would be good to throw earlier on the Add() itself.

**Comment by mrudat** [comment]
I'm not sure that the current behaviour is bad (while being modified); When constraint checking is addad, there would be still the ability to export to a generic collection, modify that (possibly violating the constraint), then write it back, where the constraint would be checked, I assume.

**Comment by Noggog** [comment]
Yeah, it's a balance.  Problem with leaving as-is is that when a mod fails to export due to having a large list, it's harder to nail down who/where was the code adding too much.   If it throws on-add, you can get a decent stack trace with hopefully a RecordException with some info, and if you run it from VS, be able to stop and see the state in the debugger.

I know a month or so ago, the KYE patcher had too many keywords added errors firing off, and trawz had to add dozens of lines of logging just to nail down which area of his code was doing it, which this would help avoid.

But yeah, the con is that you can't temporarily store more than the allowed on a list.  Maybe we can offer a specialized/customizable Add so a user can bypass if they know they're doing some add-then-remove logic?
